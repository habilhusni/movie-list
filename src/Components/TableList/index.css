.ant-table-wrapper {
  margin-top: 100px;
  margin-bottom: 100px;
  max-width: 500px !important;
  margin-left: auto;
  margin-right: auto;
}

.img-poster {
  width: 100px;
}
