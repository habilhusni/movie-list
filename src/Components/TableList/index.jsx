import React from "react";
import { Link } from "react-router-dom";
import { Table, Button } from "antd";
import "./index.css";

const columns = (prps) => [
  {
    title: "Title",
    dataIndex: "title",
    key: "title",
  },
  {
    title: "Year",
    dataIndex: "year",
    key: "year",
  },
  {
    title: "imdbID",
    dataIndex: "imdbID",
    key: "imdbID",
  },
  {
    title: "Type",
    dataIndex: "type",
    key: "action",
  },
  {
    title: "Poster",
    dataIndex: "poster",
    key: "poster",
    render: (source) => renderPoster(prps, source),
  },
  {
    title: "",
    dataIndex: "action",
    key: "action",
    render: (text, record) => renderView(prps, record),
  },
];

const renderPoster = (prps, src) => (
  <img
    onClick={() => prps.willModalShow(true, src)}
    src={src}
    alt="poster"
    className="img-poster"
  />
);

const renderView = (prps, rec) => (
  <Link
    to={{
      pathname: "/detail",
    }}
  >
    <Button onClick={() => prps.getDetails(rec.imdbID)}>View</Button>
  </Link>
);

const isMoviesAvailable = (mvs) => {
  if (!mvs) {
    return false;
  } else if (!mvs.payload) {
    return false;
  } else if (!mvs.payload.data) {
    return false;
  } else if (!mvs.payload.data.Search) {
    return false;
  } else if (!mvs.payload.data.Search.length === 0) {
    return false;
  }

  return true;
};

const data = (prps) =>
  isMoviesAvailable(prps.movies)
    ? prps.movies.payload.data.Search.map((item, i) => ({
        key: i,
        title: item.Title,
        year: item.Year,
        imdbID: item.imdbID,
        type: item.Type,
        poster: item.Poster,
      }))
    : [];

const TableList = (props) => (
  <Table
    columns={columns(props)}
    dataSource={data(props)}
    loading={props.movies.isLoading}
    pagination={false}
  />
);

export default TableList;
