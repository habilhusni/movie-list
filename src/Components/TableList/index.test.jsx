import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";

import TableList from "./index.jsx";

Enzyme.configure({ adapter: new Adapter() });

describe("Test Snapshot and Functional", () => {
  let component;
  let props = {
    movies: {
      isLoading: false,
      payload: {},
      error: {},
    },
    getDetails: () => {},
    willModalShow: () => {},
  };

  beforeEach(() => {
    component = shallow(<TableList {...props} />);
  });

  it("Testing TableList Snapshot", () => {
    expect(toJSON(component)).toMatchSnapshot();
  });
});
