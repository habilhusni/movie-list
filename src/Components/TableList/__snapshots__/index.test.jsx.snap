// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Snapshot and Functional Testing TableList Snapshot 1`] = `
<Table
  columns={
    Array [
      Object {
        "dataIndex": "title",
        "key": "title",
        "title": "Title",
      },
      Object {
        "dataIndex": "year",
        "key": "year",
        "title": "Year",
      },
      Object {
        "dataIndex": "imdbID",
        "key": "imdbID",
        "title": "imdbID",
      },
      Object {
        "dataIndex": "type",
        "key": "action",
        "title": "Type",
      },
      Object {
        "dataIndex": "poster",
        "key": "poster",
        "render": [Function],
        "title": "Poster",
      },
      Object {
        "dataIndex": "action",
        "key": "action",
        "render": [Function],
        "title": "",
      },
    ]
  }
  dataSource={Array []}
  loading={false}
  pagination={false}
  rowKey="key"
/>
`;
